### DISCUSSION TITLE
macOS Buld Failed!

### DISCUSSION THREAD

**Discussion by sorydima** [codemagic-ci-cd/codemagic-docs#2539] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Try to build macOS app with the https://github.com/sorydima/REChain-.git

### Expected results

Build done ..

### Actual results

       WebRTC-SDK (= 114.5735.10)

    You have either:
     * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
     * changed the constraints of dependency `WebRTC-SDK` inside your development pod `flutter_webrtc`.
       You should run `pod update WebRTC-SDK` to apply changes you've made.

    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:317:in `raise_error_unless_state'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:299:in `block in unwind_for_conflict'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `tap'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `unwind_for_conflict'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:257:in `process_topmost_state'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:182:in `resolve'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/molinillo-0.8.0/lib/molinillo/resolver.rb:43:in `resolve'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/resolver.rb:94:in `resolve'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer/analyzer.rb:1078:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer/analyzer.rb:1076:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer/analyzer.rb:124:in `analyze'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer.rb:416:in `analyze'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer.rb:241:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer.rb:240:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/installer.rb:161:in `install!'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/command/install.rb:52:in `run'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:52:in `run'
    /Users/builder/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/cocoapods-1.11.3/bin/pod:55:in `<top (required)>'
    /Users/builder/.rbenv/versions/2.7.2/bin/pod:25:in `load'
    /Users/builder/.rbenv/versions/2.7.2/bin/pod:25:in `<main>'

Error output from CocoaPods:
↳
    [!] The version of CocoaPods used to generate the lockfile (1.14.3) is higher than the version of the current executable (1.11.3). Incompatibility issues may arise.

Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.
To update the CocoaPods specs, run:
  pod repo update

Error running pod install

Build failed :|
Failed to build for macOS

Build failed :|
Failed to build for macOS

![Снимок экрана от 2024-05-17 19-04-04](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/9bc75844-5ebd-4099-bb5c-c20906b5f485)
![Снимок экрана от 2024-05-17 19-04-10](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/de3ea0ee-63e3-46a8-a42a-bf5ace01e0b6)
![Снимок экрана от 2024-05-17 19-04-18](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/1dda5685-ae46-4ab1-95d1-d9693c55e2f1)

### Build id (optional)

66477f146708dc6bf34223ba

**Comment by dtrdic** [comment]
Hello @sorydima, i can see that you have resolved the above error and you are now facing a different one.

`​"The provided entity includes an attribute with a value that has already been used. The bundle version must be higher than the previously uploaded version"`
​
This error typically occurs because you've previously uploaded an app version with the same bundle version to the App Store.
I also see that you have already added the build name and build number arguments with a built-in environment variable which is automatically incremented with each new build, that's ok. So you probably hardcoded your app version inside pubspec.yaml.
Make sure your app version looks like this inside pubspec.yaml.
<img width="357" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/2fdca140-519a-4ea0-8f84-40fcf78be1e7">

Also make sure that Info.plist file has the following string along with their key values

```
<key>CFBundleShortVersionString</key>
<string>$(FLUTTER_BUILD_NAME)</string>
<key>CFBundleVersion</key>
<string>$(FLUTTER_BUILD_NUMBER)</string>
```

**Reply by sorydima** [reply]
So on and on and..)) 💬
Now!
Failure: POST https://api.appstoreconnect.apple.com/v1/reviewSubmissionItems returned 409: appStoreVersions with id 'e600ebd1-d21c-4d15-a49a-ff7d4ddf0160' is not in valid state. - This resource cannot be reviewed, please check associated errors to see why.
Associated error: The provided entity includes a relationship with an invalid value - Platform for version e600ebd1-d21c-4d15-a49a-ff7d4ddf0160 should match the platform for review submission.

https://codemagic.io/app/6640926573b707f95546cea0/build/664d9ef7665fc2f673d67cd6


![Screenshot_20240522-132956.png](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/4648d201-1020-4eac-b1b2-e3fde844423b)

Any thoughts?) 💭

**Reply by dtrdic** [reply]
This error can occur when using an invalid version string for your bundle version, like `1.0.６` instead of `1.0.6`
I see you now removed the build arguments from your build command, please add it again.
`--build-name=1.0.$PROJECT_BUILD_NUMBER --build-number=1.0.$PROJECT_BUILD_NUMBER`

Also make sure that Info.plist file has the following string along with their key values
```
<key>CFBundleShortVersionString</key>
<string>$(FLUTTER_BUILD_NAME)</string>
<key>CFBundleVersion</key>
<string>$(FLUTTER_BUILD_NUMBER)</string>
```
Also check [this SO discussion](https://stackoverflow.com/questions/74163395/the-provided-entity-includes-a-relationship-with-an-invalid-value) for reference

**Reply by sorydima** [reply]
Failure: POST https://api.appstoreconnect.apple.com/v1/reviewSubmissionItems returned 409: appStoreVersions with id 'e600ebd1-d21c-4d15-a49a-ff7d4ddf0160' is not in valid state. - This resource cannot be reviewed, please check associated errors to see why.
	Associated error: The provided entity includes a relationship with an invalid value - Platform for version e600ebd1-d21c-4d15-a49a-ff7d4ddf0160 should match the platform for review submission.

Check, please 🥺 

https://codemagic.io/app/6640926573b707f95546cea0/build/664ddb621588ccb4f83931d4

**Reply by dtrdic** [reply]
If you expand the publishing section you will see
```
Version code: 1142
Version: 4.0.0
```
you have hardcoded this version somewhere, have you modified your pubspec.yaml as i suggested above?

**Reply by sorydima** [reply]
Could you check the repo, please 🥺 
So what should I do?

**Reply by dtrdic** [reply]
Please change [this](https://github.com/sorydima/REChain-/blob/main/pubspec.yaml#L4) to version: 1.0.0+1
And change [this](https://github.com/sorydima/REChain-/blob/main/ios/Runner/Info.plist#L43) to `<string>$(FLUTTER_BUILD_NUMBER)</string>`

**Reply by sorydima** [reply]
"NSLocalizedDescription": "The provided entity includes an attribute with a value that has already been used",
                "previousBundleVersion": "1142",
                "iris-code": "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE",
                "NSLocalizedFailureReason": "The bundle version must be higher than the previously uploaded version: \u20181142\u2019. (ID: 2cd9a40f-903d-4009-a389-939e0bff7716)"
            },
            "code": -19232
        }
    ]
}
Failed to upload archive at "/Users/builder/clone/build/macos/Build/Products/Release/REChain.pkg"
Failed to publish /Users/builder/clone/build/macos/Build/Products/Release/REChain.pkg

Failed to publish REChain.pkg to App Store Connect.

Build failed :|


Publishing failed :|
Failed to publish REChain.pkg to App Store Connect.


https://codemagic.io/app/6640926573b707f95546cea0/build/664df78ea334e6038462b5ee

**Comment by dtrdic** [comment]
Note: Continued and resolved over Intercom chat
